### ISSUE TITLE
Decompiler: raise pattern-guarded short-circuit && diamond, inline default temp, and re-compose fluent chain (BePositive residue)

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#2877]
Follow-up raising residue observed on the `#2872`/`#2873` witness member `FluentAssertions.Numeric.NumericAssertionsBase\`3::BePositive` (net6.0). After #2873 recovers the `is T subject` cache-bind, the method is correct but still not fully raised relative to authored source.

**Authored source** (SourceLink, via `dotnet-inspect member NumericAssertionsBase BePositive --package FluentAssertions -S "Original Source"`):

```csharp
CurrentAssertionChain
    .ForCondition(Subject is T subject && subject.CompareTo(default) > 0)
    .BecauseOf(because, becauseArgs)
    .FailWith("Expected {context:value} to be positive{reason}, but found {0}.", Subject);
```

**Current decompiled output:**

```csharp
AssertionChain S_256 = CurrentAssertionChain;
if (Subject is T subject)
{
    V_2 = default;
    S_1 = subject.CompareTo(V_2) > 0;
}
else
{
    S_1 = false;
}
object[] S_257 = new object[1];
AssertionChain S_258 = S_256.ForCondition(S_1).BecauseOf(because, becauseArgs);
S_257[0] = Subject;
S_258.FailWith("Expected {context:value} to be positive{reason}, but found {0}.", S_257);
```

## Remaining slices

1. **Short-circuit `&&` reconstruction.** The `if (Subject is T subject) { S_1 = <rhs>; } else { S_1 = false; }` boolean-assignment diamond should raise to `Subject is T subject && <rhs>` (a pattern-guarded short-circuit `&&` producing a bool). This is the general "boolean if/else-to-`&&`/`||` short-circuit expression" raise, specialized here to a declaration-pattern left operand.
2. **`default` literal inlining.** `V_2 = default; ... subject.CompareTo(V_2)` should inline to `subject.CompareTo(default)` (single-assignment temp holding a literal, read once).
3. **Fluent chain / temp re-composition.** The `S_256`/`S_257`/`S_258` receiver and params-array temps should collapse back to the fluent `.ForCondition(...).BecauseOf(...).FailWith(..., Subject)` chain.

Reaching all three yields the authored shape. Tracking here so #2873 can record the intended endpoint without overclaiming "fully raised".
